docs/demos/demos/intro_demo.py

```diff
@@ -0,0 +1,256 @@
+#
+# In the example below, the circuit contains a single qubit register with two
+# qubits; it could contain any number of qubit registers. You can
+#
+# * add another register with the `Circuit.add_qregister` method, where an argument `n`
+#   is the number of qubits in the new register
+# * add a qubit with `Circuit.add_qubit`, optionally passing a qubit object
+#   and/or a register to which to add it.
+
+circuit = Circuit(2)
+
+with circuit.context as reg:
+    ops.X(reg.q[0])
+
+###############################################################################
+# This example created a circuit object with two qubits in its register, applying
+# a single X gate to the first qubit. Print the circuit to see general information
+# about it: type of circuit, number of qubits/bits, and number of operations.
+
+print(circuit)
+
+###############################################################################
+# ## Applying gates to circuits
+#
+# You can apply operations to a circuit in several different ways, as demonstrated
+# in the example below. You can pass both qubits and parameters as either single
+# values (when supported by the gate) or sequences. Note that different types of
+# gates accept slightly different arguments, although you can _always_ pass the
+# qubits as sequences via the keyword argument `qubits`.
+#
+# :::note
+# Always apply any operations you instantiate within a circuit context to
+# specific qubits in the circuit's qubit register. You can access the qubit
+# register via the named tuple returned by the context manager as `q`, indexing
+# into it to retrieve the corresponding qubit.
+# :::
+
+circuit = Circuit(3)
+
+with circuit.context as (q, c):
+    # apply single-qubit gate
+    ops.Z(q[0])
+    # apply single-qubit gate using kwarg
+    ops.Hadamard(qubits=q[0])
+    # apply a parametric gate
+    ops.RY(4.2, q[1])
+    # apply a parametric gate using kwargs
+    ops.Rotation(parameters=[4.2, 3.1, 2.3], qubits=q[1])
+    # apply controlled gate
+    ops.CNOT(q[0], q[1])
+    # apply controlled gate using kwargs
+    ops.CX(control=q[0], target=q[1])
+    # apply controlled qubit gates using slicing (must unpack)
+    ops.CZ(*q[:2])
+    # apply multi-qubit (non-controlled) gates (note tuple)
+    ops.SWAP((q[0], q[1]))
+    # apply multi-qubit (non-controlled) gates using kwargs
+    ops.CSWAP(qubits=(q[0], q[1], q[2]))
+    # apply multi-qubit (non-controlled) gates using slicing
+    ops.SWAP(q[:2])
+    # apply gate on all qubits in the circuit
+    ops.Toffoli(q)
+
+###############################################################################
+# You can access all the operations in a circuit using the `Circuit.circuit`
+# property. The code below iterates over the returned list of all operations that
+# have been applied to the circuit.
+
+for op in circuit.circuit:
+    print(op)
+
+###############################################################################
+# :::note
```

Comment:
Does not render:

![image](https://github.com/dwavesystems/dwave-gate/assets/34041130/ca82e9b3-3d27-47cb-a2fc-d63b70e0c721)